pyomo/contrib/pynumero/interfaces/cyipopt_interface.py

```diff
@@ -309,6 +310,38 @@ def __init__(self, nlp, intermediate_callback=None, halt_on_evaluation_error=Non
         # cyipopt.Problem.__init__
         super(CyIpoptNLP, self).__init__()
 
+        self._use_13arg_callback = None
+        if self._intermediate_callback is not None:
+            signature = inspect.signature(self._intermediate_callback)
+            positional_kinds = {
+                inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                inspect.Parameter.POSITIONAL_ONLY,
+            }
+            positional = [
+                param
+                for param in signature.parameters.values()
+                if param.kind in positional_kinds
+            ]
+            has_var_args = any(
+                p.kind is inspect.Parameter.VAR_POSITIONAL
+                for p in signature.parameters.values()
+            )
+
+            if len(positional) == 13 and not has_var_args:
+                # If *args is expected, we do not use the new callback
+                # signature.
+                self._use_13arg_callback = True
+            elif len(positional) == 12 or has_var_args:
+                # If *args is expected, we use the old callback signature
+                # for backwards compatibility.
+                self._use_13arg_callback = False
+            else:
+                raise ValueError(
+                    "Invalid intermediate callback. A function with either 12"
+                    "or 13 positional arguments, or a variable number of arguments,"
```

Comment:
You'll need these spaces. Otherwise it'll print out oddly.
```suggestion
                    "Invalid intermediate callback. A function with either 12 "
                    "or 13 positional arguments, or a variable number of arguments, "
```